workshops/reddit_viewer/README.md

```diff
@@ -0,0 +1,126 @@
+
+Currently we do not have a "submissions" variable. That is easy to create. Go to the main.py file and in your return statement in the \_post method add a comma after the index.html and add this line. `submissions=reddit.subreddit(subreddit.upper()).hot(limit=100)`. This is just saying that the submissions variable in the html should be all the posts from the user defined subreddit (which is regardless of case because of the .upper). Then we sort them by hot, and only showing the top 100 posts.
+
+That is all we need for the python part. Now we just need to finish off the html. In your for-statement section in the html file, make an `<a>` element with an `href={{ "https://reddit.com" + submission['permalink']}}`. This just means that we want the title to go to the actual reddit post with comments when we click on it. We must set it with reddit.com or else it will try to use whatever website we are currently on.
+
+Now in the `<a>` section add an `<h2>` that in between it is `{{ submission['title']. This just means that we can see the posts title.
+
+Now under the `<a>` add an img tag and make its `src={{ submission['url']`. This is because the URL of the post we are getting is also the image.
+
+Now run your program and you should be able to enter a subreddit name and see its posts.
+
+# Step 6: Adding some styling
+
+This step is up to you to do, but I will be centering the site and making the images fit on the page.
+
+After your body tag add a `<style>` tag. Now make a body selector.
+
+To center the site, I found that this setup worked best:
+`align-content: center;`
+`text-align: center;`
+And to make the images fit, make an img selector and make the width and height 50%.
+
+Now you have made the reddit reader! Customize it to your will and maybe you can make a whole reddit website
+
+# Final Details and Extra Comments
+
+The main.py file
+
+![](https://cloud-mephnkqgy.vercel.app/image.png)
+
+The final HTML
+
+![](https://cloud-fqz8oxjxr.vercel.app/image.png)
+
+I would highly reccomend reading more of the [PRAW library documentation](https://praw.readthedocs.io/en/latest/). Also learning Flask is a great idea for any developer working with Python. Google is full of tutorials on how to use Flask and there are many projects that are easier done in Flask.
+
+[FreeCodeCamp Flask Tutorial](https://www.youtube.com/watch?v=Z1RJmh_OqeA)
+
+[The FullStackPython written guide on Flask](https://www.fullstackpython.com/flask.html)
+
+[TutorialsPoints' Complete overview of Flask](https://www.tutorialspoint.com/flask/index.htm)
+
+[A very big guide on not just Flask but also using it](https://blog.miguelgrinberg.com/post/the-flask-mega-tutorial-part-i-hello-world)
+
+Thanks for reading!
+--Kai
+
+[In video form](https://cloud-djph69ehn.vercel.app/zoom_0.mp4)
```

Comment:
Don’t include the video link in the workshop—that’s just for the submission on GitHub :)

Reply:
Done!